snapcraft/internal/repo/_deb.py

```diff
@@ -209,20 +211,79 @@ def install_build_packages(cls, package_names):
 
     @classmethod
     def _mark_install(cls, apt_cache, package_names):
-        for name in package_names:
-            if name.endswith(':any'):
-                name = name[:-4]
+        for pkg in package_names:
+            logger.debug('Marking {!r} (and its dependencies) to be '
+                         'fetched'.format(pkg))
+            name, arch, version = repo.get_pkg_name_parts(pkg)
             if apt_cache.is_virtual_package(name):
                 name = apt_cache.get_providing_packages(name)[0].name
-            logger.debug('Marking {!r} (and its dependencies) to be '
-                         'fetched'.format(name))
-            name_arch, version = repo.get_pkg_name_parts(name)
             try:
                 if version:
-                    _set_pkg_version(apt_cache[name_arch], version)
-                apt_cache[name_arch].mark_install()
-            except KeyError:
-                raise errors.PackageNotFoundError(name)
+                    _set_pkg_version(apt_cache[name + arch], version)
+                apt_cache[name + arch].mark_install()
+            except KeyError as e:
+                raise errors.PackageNotFoundError(pkg) from e
+
+    @classmethod
+    def _get_build_deps(cls, package_names, arch):
+        """Use apt-get build-dep with a fake source package file to find
+           all dependencies and correct architectures.
+        """
+        build_deps = []
+        if not package_names:
+            return build_deps
+
+        with tempfile.NamedTemporaryFile(suffix='.dsc') as fake_source:
+            depends = 'Build-Depends: {}\n'.format(', '.join(package_names))
```

Comment:
Can you explain how this has different results than using something like [apt_pkg.parse_depends()](https://apt.alioth.debian.org/python-apt-doc/library/apt_pkg.html#apt_pkg.parse_depends)?